### ISSUE TITLE
Error using collapsed sub process

### ISSUE THREAD

**Issue by callaertanthony** [camunda-community-hub/camunda-process-test-coverage#259]
Hello,

Using coverage with a collapsed sub process doesn't displays full coverage even if the process is full covered.
An error is thrown in the console :

```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'id')
    at ln._updateMarker (Canvas.js:482:55)
    at ln.addMarker (Canvas.js:531:8)
    at BpmnViewer.tsx:163:32
    at Array.forEach (<anonymous>)
    at BpmnViewer.tsx:162:46
    at c (runtime.js:63:40)
    at Generator._invoke (runtime.js:293:22)
    at Generator.next (runtime.js:118:21)
    at r (asyncToGenerator.js:3:20)
    at s (asyncToGenerator.js:25:9)
```

<img width="367" alt="Capture d’écran 2022-10-18 à 14 40 41" src="https://user-images.githubusercontent.com/5382102/196431980-f6bf8099-006e-4f58-a017-9f7421ea5385.png">

My current version is `camunda-process-test-coverage-junit5-platform-7:2.0.0`

**Comment by rohwerj** [comment]
@callaertanthony could you please provide the report.html and report.json, that can reproduce the error?

**Comment by callaertanthony** [comment]
Hi,

please find the [report.html](https://gist.github.com/callaertanthony/a78feb726ee2337aa309dc4305d7af34) and the [report.json](https://gist.github.com/callaertanthony/ac592631582839176e3802a0e7f7697d)

**Comment by alex-praschek** [comment]
Hi, 

sorry for the delayed response, I somehow confused this issue with another one and thought it had already been resolved. I took a closer look at this and it seems as if there are two issues being mixed up here.

### Issue 1

The first issue is the exception in the logs. It is being thrown because we try to add a marker (i.e. background color) to an element inside the collapsed subprocess. However, bpmn-js only added support for collapsed subprocesses in v9.0.0, while we are still using v8.8.2 (see [changelog](https://github.com/bpmn-io/bpmn-js/blob/develop/CHANGELOG.md)). Therefore, the exception is being thrown and prevents the model from being rendered correctly. If I update this to the latest version (v10.3.0), these exceptions stop. However, there is another issue that starts to appear after the update:

```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'length')
    at BpmnViewer.tsx:208:1
```

I will create a PR for the flowcov-viewer repository. Once this is done, we can update this library as well.

### Issue 2

@callaertanthony If I understand you correctly, you are expecting the coverage to be at 100%. However, the coverage is actually at 95,65% (44/46 nodes covered). 

![image](https://user-images.githubusercontent.com/5673767/203557533-ed27d298-c035-4938-ac8d-bd98e2ea4484.png)

I took a closer look where this discrepancy is coming from and compared the list of all elements in the model with the list of covered elements. There are two elements that are not being covered by the tests:

```
Event_1uiqgfb
Flow_0ne813s
```

If you check the model, these are the ids of the boundary event and the outgoing sequence flow in the second collapsed subprocess ("close order"). If you compare this to the coverage report, these are indeed not covered by the tests (see the background color of the boundary event):

> This only works after updating the version (see Issue 1).

![image](https://user-images.githubusercontent.com/5673767/203558124-2f71a09d-3865-44fd-9df0-dc2f8c26a925.png)

Therefore, the calculated coverage is correct. Since I don't have access to the tests, I can't really tell if they are *actually* covered or not. But since the other boundary events seem to work perfectly fine, I don't think that this is an issue.

I hope this helps. 

Best regards,
Alex

**Comment by callaertanthony** [comment]
Hi @alex-praschek !
I'm not expecting the coverage to be at 100%. My test class doesn't cover the two elements mentioned behind.
My issue was only based on the fact you mentioned in your first issue in your comment.

Regards

**Comment by rohwerj** [comment]
Hi @alex-praschek.
Is there an update regarding the status of this issue?
Was the library updated or are there still open issues?

**Comment by alex-praschek** [comment]
Hi, thank you for reminding me. We're tracking this in https://github.com/FlowSquad/flowcov-viewer/issues/6. The issue has not been resolved yet, sorry.

**Comment by rohwerj** [comment]
@rami-nk Can confirm it fixes the problem. Thanks a lot.
